pkg/controller/infinispan/resources/operatorconfig/grafana.go

```diff
@@ -0,0 +1,80 @@
+package operatorconfig
+
+import (
+	"context"
+
+	"github.com/infinispan/infinispan-operator/pkg/controller/constants"
+	consts "github.com/infinispan/infinispan-operator/pkg/controller/constants"
+	"github.com/infinispan/infinispan-operator/pkg/controller/infinispan"
+	grafanav1alpha1 "github.com/integr8ly/grafana-operator/v3/pkg/apis/integreatly/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	grafanaDashboardNamespace = "grafanaDashboardNamespace"
+)
+
+func (r *ReconcileOperatorConfig) reconcileGrafana(config map[string]string, operatorNs string) (*reconcile.Result, error) {
+	grafanaNs := config[grafanaDashboardNamespace]
+	if grafanaNs != "" {
+		// detect the GrafanaDashboard resource type resourceExists on the cluster
+		resourceExists, err := r.Kube.IsGroupVersionSupported(grafanav1alpha1.SchemeGroupVersion.String(), grafanav1alpha1.GrafanaDashboardKind)
+		if err != nil {
+			r.Log.Error(err, "Error checking Grafana support")
+			return &reconcile.Result{Requeue: true}, nil
+		}
+		if !resourceExists {
+			r.Log.Info("Grafana CRD not present - not installing dashboard CR")
+			return &reconcile.Result{RequeueAfter: consts.DefaultWaitOnCreateResource}, nil
+		}
+		infinispanDashboard := dashboard(grafanaNs)
+		operatorDeployment := &appsv1.Deployment{}
+		result, err := infinispan.LookupResource("infinispan-operator", operatorNs, operatorDeployment, r.client, r.Log)
+		if err != nil {
+			r.Log.Error(err, "There was a problem looking up the infinispan operator to add the dashboard as a secondary resource")
+			return &reconcile.Result{Requeue: true}, nil
+		}
+		if result == nil {
+			if err = controllerutil.SetControllerReference(operatorDeployment, infinispanDashboard, r.scheme); err != nil {
+				r.Log.Error(err, "Unable to set Grafana Dashboard controller reference to Infinispan operator - skipping creation")
+				return &reconcile.Result{Requeue: true}, nil
+			}
+		} else {
+			r.Log.Info("Infinispan operator resource not found, assuming operator is running locally.")
+		}
+		if _, err = controllerutil.CreateOrUpdate(context.Background(), r.client, infinispanDashboard, func() error {
```

Comment:
we can use `ctx` here.

Reply:
As we're not using the `mutate` function, can we just call `client.Create` here?